src/Register/actions.js

```diff
@@ -0,0 +1,39 @@
+import REGISTER from './action_types';
+import query from '../util/query';
+
+export function setUsername(value) {
+    return {
+        type: REGISTER.USERNAME,
+        value
+    };
+}
+
+export function setPassword(value) {
+    return {
+        type: REGISTER.PASSWORD,
+        value
+    };
+}
+
+export function setFullName(value) {
+    return {
+        type: REGISTER.FULLNAME,
+        value
+    };
+}
+
+export function submit() {
+    return {
+        type: REGISTER.SUBMIT
+    };
+}
+
+export function attemptRegister(name, password, fullName) {
+    return async function (dispatch) {
+        const res = await query({query: `mutation {
+            createUser(data:{uid:'${name}', password:'${password}', fullName:'${fullName}'})
```

Comment:
we should use format like this if we need to pass in variables.
{
   query: ... 
   variables: ...
}